src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs

```diff
@@ -459,9 +459,20 @@ public override int GetHashCode()
 
     internal class PathStringConverter : TypeConverter
     {
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+            => sourceType == typeof(string) 
+            ? true 
+            : base.CanConvertFrom(context, sourceType);
+
         public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
-        {
-            return new PathString((string)value);
-        }
+            => value is string 
+            ? new PathString((string)value) 
```

Comment:
This replicates the asymmitry bug from the implicit converters:
https://github.com/aspnet/HttpAbstractions/issues/690

string -> PathString should use FromUriComponent(string), not new PathString(string)

We should change them both at the same time. @davidfowl